### ISSUE TITLE
Solaredge branch - Zero-feed in limit stuck at 1064 W for Solaredge SE 66.6 K (Synergy manager)

### ISSUE THREAD

**Issue by kyros32** [victronenergy/dbus-fronius#7]
@izak @mpverrijt

#https://github.com/mpverrijt/dbus-fronius
#6 https://github.com/victronenergy/dbus-fronius/issues/6

Hello, since this repo by @mpverrijt was pulled here to the solaredge branch, I think it is best if we keep it maintained here.
I have commented on a closed issue here, but I think its best to open a new one.

**The problem:**
I have commissioned 3x Multiplus 15 kVA + Solaredge SE 66.6 K Synergy (AC coupled on the AC INPUT). Normally, ie. with Solaredge SE 17 K power throttling works flawlessly. However, in my case not - it is stuck at 1064 W. When I dbus-spy it, I don't see anything weird except this. So I think there must be some error or incompatibility in the code - maybe wrong detection of the inverter? I tried to build with 3.52 SDK this repository https://github.com/mpverrijt/dbus-fronius and also this https://github.com/victronenergy/dbus-fronius/tree/solaredge with the correct branch. The outcome is always the same.

I tried to check the code but I am not an expert, so I cant find

Please see attached screens with dbus-spy. Any help is much appreciated. I really don't want to deploy a PLC for this task :-/ Thank you very much.

<img width="819" alt="Snímek obrazovky 2024-12-25 v 22 49 46" src="https://github.com/user-attachments/assets/854e0457-55b5-4ec4-acae-7ec85305d79b" />
<img width="837" alt="Snímek obrazovky 2024-12-25 v 22 50 14" src="https://github.com/user-attachments/assets/fd65c3ee-0442-4a09-ae26-ffc91bd2c32f" />
<img width="524" alt="Snímek obrazovky 2024-12-24 v 22 47 50" src="https://github.com/user-attachments/assets/282c2f52-c0e0-4197-8aac-d2607cac0f47" />

**Comment by kyros32** [comment]
I actually think it might be related to incorrect data type handling / endianess of 0xF304 which should be float?

**Comment by mpverrijt** [comment]
Seems to me as an uint16 rollover. 2^16 (65536) + 1064 = 66600.

First it tries to get the maxPower via "702: // IEEE 1547 DERCapacity page" if it exists as uint16:
```cpp
		case 702: // DERCapacity, new IEEE 1547 alternative for 120
			if (values.size() > 45)
				di->di.maxPower = getScaledValue(values, 2, 1, 45, false);
			break;
```

See also this [comment](https://github.com/victronenergy/dbus-fronius/issues/6#issuecomment-2373572682) for more info. 

I just made a quick test version of the solaredge branch in the fork i made (I haven't got around to testing the merged solaredge code in this repo yet). It will always get the value from the 0xF304 register for maxPower. See the fork+branch [here](https://github.com/mpverrijt/dbus-fronius/tree/debug/kyros32_test).

**Comment by kyros32** [comment]
Hi @mpverrijt , thank you! You're THE man! Victron Energy should employ you!. If you are already an employee of VE, you should get a raise! ;-) It works correctly now. I will report if there are any issues.


<img width="499" alt="Snímek obrazovky 2024-12-29 v 13 14 38" src="https://github.com/user-attachments/assets/5deec415-3aac-4556-8bf0-fc25e8702e0c" />
<img width="494" alt="Snímek obrazovky 2024-12-29 v 13 14 45" src="https://github.com/user-attachments/assets/d20e2420-058d-471b-b502-6281029c984c" />

**Comment by mpverrijt** [comment]
@izak: So the fallback I made for older inverters (without 120/702 model), which then gets the maxPower value from register 0xF304 as float32 can always be applied for the solaredge inverters. This is what I did for @kyros32  to test with (in the "old" fork). In the merged code that is easily adapted by removing the if condition [here](https://github.com/victronenergy/dbus-fronius/blob/2c7cf79c7a37a37de83a088dd95602bef4ef9e47/software/src/solaredge_limiter.cpp#L47) if you agree this is a reasonable solution.

**Comment by izak** [comment]
@mpverrijt Indeed, a workaround is possible, and I am not against that at all. Thanks for doing the hard lifting on this!

If we ever end up at a table with a few beers between us, I can tell you all about the various ways manufacturers murder SunSpec and make my life difficult :-)

**Comment by mpverrijt** [comment]
@izak I created a pull request for the solardege branch https://github.com/victronenergy/dbus-fronius/pull/8

Just curious, when do you plan to merge the solaredge branch into the master branch, after we have tested it (which I haven't yet)? or are there more/other procedures that need to be done by Victron aswell?

**Comment by izak** [comment]
The important question here is whether this change might break anything.

The old behaviour skips fetching that register for some inverters. So the question is, are there SolarEdge inverters that doesn't have this value, so that if this change is made they will no longer work?

Mainly, the reason for the delay is simply that I've been working on other things. The change is simple and if you are confident that we won't be breaking anything, I can merge it almost immediately.

**Comment by mpverrijt** [comment]
@izak, thats great news that it's in the main branch. From the 3 inverters I could test on this register always had the correct value. I will soon be enabling dynamic ESS myself as I am now going to a dynamic contract. I will update to the latest main branch version before doing this.
